debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerChangeTableSetIT.java

```diff
@@ -185,25 +185,39 @@ public void removeTable() throws Exception {
     }
 
     @Test
-    public void addColumnToTableEndOfBatch() throws Exception {
-        addColumnToTable(true);
+    public void addColumnToTableEndOfBatchWithoutLsnLimit() throws Exception {
+        addColumnToTable(true, false);
     }
 
     @Test
-    public void addColumnToTableMiddleOfBatch() throws Exception {
-        addColumnToTable(false);
+    public void addColumnToTableEndOfBatchWithLsnLimit() throws Exception {
+        addColumnToTable(true, true);
     }
 
-    private void addColumnToTable(boolean pauseAfterCaptureChange) throws Exception {
+    @Test
+    public void addColumnToTableMiddleOfBatchWithoutLsnLimit() throws Exception {
+        addColumnToTable(false, false);
+    }
+
+    @Test
+    public void addColumnToTableMiddleOfBatchWithLsnLimit() throws Exception {
+        addColumnToTable(false, true);
+    }
+
+    private void addColumnToTable(boolean pauseAfterCaptureChange, boolean limitTransactionsPerIteration) throws Exception {
         final int RECORDS_PER_TABLE = 5;
         final int TABLES = 2;
         final int ID_START_1 = 10;
         final int ID_START_2 = 100;
         final int ID_START_3 = 1000;
         final int ID_START_4 = 10000;
-        final Configuration config = TestHelper.defaultMultiDatabaseConfig()
-                .with(SqlServerConnectorConfig.SNAPSHOT_MODE, SnapshotMode.SCHEMA_ONLY)
-                .build();
+
+        final Configuration.Builder builder = TestHelper.defaultMultiDatabaseConfig()
+                .with(SqlServerConnectorConfig.SNAPSHOT_MODE, SnapshotMode.SCHEMA_ONLY);
+        if (limitTransactionsPerIteration){
```

Comment:
Please avoid flow control in tests. Instead, have each test prepare its relevant part of the config and have this method build the common part. See the rework of `SnapshotWithSelectOverridesIT#takeSnapshotWithOverrides` in https://github.com/debezium/debezium/pull/2604 for example.